app/models/ip.rb

```diff
@@ -12,8 +12,12 @@ def available?
 
   def address_must_be_valid_ip
     checker = UseCases::Administrator::CheckIfValidIp.new
-    valid_ip = checker.execute(self.address)[:success]
+    valid_ipv4 = checker.execute(self.address)[:success]
+    valid_ipv6 = checker.execute(self.address)[:ipv6?]
```

Comment:
Calling `checker.execute` twice is a bit of a code smell. Try refactoring so that it only needs to be called once.